File: engine/src/main/java/edu/austral/dissis/chess/engine/Piece.java
package edu.austral.dissis.chess.engine;


public class Piece {
  private Color color;
  private int initialPosition;
  public int actualPosition;
  private boolean isAlive;
  private boolean hasMoved;

  public Piece(Color color, int initialPosition, int actualPosition, boolean isAlive, boolean hasMoved) {
    this.color = color;
    this.initialPosition = initialPosition;
    this.actualPosition = actualPosition;
    this.isAlive = isAlive;
    this.hasMoved = hasMoved;
  }

  public boolean isAlive() {
    return isAlive;
  }
  public Color getColor() {
    return color;
  }
  public int getActualPosition() {
    return actualPosition;
  }
}

File: engine/src/main/kotlin/edu/austral/dissis/chess/engine/Piece/Piece.kt
package edu.austral.dissis.chess.engine.Piece

class Piece(val initialPosition: Position, val actualPosition: Position, val pieceColor: Color,
            val type: PieceType, val hasMoveed: Boolean, val movements: List<Position>, val id: String) {

  fun getInitialPosition(): Position {
    return initialPosition
  }

  fun getActualPosition(): Position {
    return actualPosition
  }

  fun getColor(): Color {
    return pieceColor
  }

  fun getType(): PieceType {
    return type
  }

  fun getMovements(): List<Position> {
    return movements
  }

  fun getId(): String {
    return id
  }
}